### PR TITLE
make directory copies idempotent

### PR DESCRIPTION
**Pull request by biosugar0**
Thanks for the great project.
I have made one improvements to directory copy command.

The copy command `cp -R ${from} ${to}` creates an extra directory in the destination directory if it already exists.
```
to
├── main.tf
└── from
    └── main.tf
```
So I changed the command to overwrite it.
`cp -R ${from}/. ${to}`
```
to
└── main.tf
```
I prefer this behavior.

### CODE REVIEW

**Review by suzuki-shunsuke — COMMENTED** [review]
Thank you for your contribution!
You'd like to create and update files and directories in an existing directory, right?
I haven't thought such a usecase, but it may be useful when developers create working directories without the scaffold workflow.
Sometimes they do that because they don't know tfaction well.

Let me check if it works well, because I have never thought such a usecase.

**Comment by biosugar0** [comment]
Yes, it is. It is useful for updating. Additionally, it is beneficial in terms of automation.

In my use case, in a development environment, where we often replicate a particular infrastructure set, 
we create a pull request in the scaffold workflow, which is automatically merged after terraform plan and terraform apply.

If we mistakenly trigger the workflow a second time, a directory is created and merged that we do not need, so it is preferable that the copy is overwritten with the most recent template directory.

Additionally, we are experimenting with triggering workflow_dispatch by k8s operator, which may result in a duplicate run.

**Comment by suzuki-shunsuke** [comment]
> In my use case, in a development environment, where we often replicate a particular infrastructure set,
> we create a pull request in the scaffold workflow, which is automatically merged after terraform plan and terraform apply.
> 
> If we mistakenly trigger the workflow a second time, a directory is created and merged that we do not need, so it is preferable that the copy is overwritten with the most recent template directory.
> 
> Additionally, we are experimenting with triggering workflow_dispatch by k8s operator, which may result in a duplicate run.

Thank you for your explanation.
In this case, it may be better that CI fails when the directory already exists.

**Comment by biosugar0** [comment]
Yes, there may be such a case.
If we make a directory fail when it already exists, we want to use output to determine whether the failure is because the directory exists or some other error.

Alternatively, we may simply skip the case where the directory already exists.

Currently, an unintended directory is created and exits normally.

File: scaffold-working-dir/action.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ runs:
     - run: mkdir -p "$(dirname "${{steps.target-config.outputs.working_directory}}")"
       shell: bash
 
-    - run: cp -R "${{steps.target-config.outputs.template_dir}}" "${{steps.target-config.outputs.working_directory}}"
+    - run: cp -R "${{steps.target-config.outputs.template_dir}}/." "${{steps.target-config.outputs.working_directory}}"
```
Comment (biosugar0) [inline scaffold-working-dir/action.yaml:21]: skip case. this is more better?

```suggestion
    - run: |
        if [ -d "${{steps.target-config.outputs.working_directory}}" ]; then
          cp -R "${{steps.target-config.outputs.template_dir}}" "${{steps.target-config.outputs.working_directory}}"
        else
          :
        fi
```

Comment (biosugar0) [reply scaffold-working-dir/action.yaml:21]: In this way, the workflow create pull request with `no-changes` label. It seems like good.

**Comment by suzuki-shunsuke** [comment]
Sorry for late reply.

When the directory already exists, there are some options.

1. Fail the step
1. Fail the step with an output to distinguish the cause of failure
1. Skip the step
1. Overwrite files
1. Copy only files and directories that don't exist: 👎  It is a little difficult to implement
1. Keep the current implementation: 👎  The current implementation isn't good, so we should improve somehow

We can decide one option or allow users to select the option as action input.

I'm considering what to do.

File: scaffold-working-dir/action.yaml (hunk 1 above)
Comment (suzuki-shunsuke) [reply scaffold-working-dir/action.yaml:21]: The `if` condition seems to be wrong.

AS IS

```sh
if [ -d "${{steps.target-config.outputs.working_directory}}" ];
```

TO BE

```sh
if [ ! -d "${{steps.target-config.outputs.working_directory}}" ];
```

And `else` isn't needed.

Comment (biosugar0) [reply scaffold-working-dir/action.yaml:21]: That's right ! fixed it.

**Comment by biosugar0** [comment]
I prefer option 3 to skip, but I will leave it up to you.
If you choose another option and no longer need this pull request, please close it 👍

**Pull request closed without merging**